### PR TITLE
Update generated gemspec

### PR DESCRIPTION
**Pull request by dentarg**
I ran "bundle exec rake gem:gemspec" and this is what changed, except
the date (did not change the date as I think that should reflect when
releases are made).

Related to #460 and #463 that both updated the generated gemspec
directly. Includes the change from #425.

### CODE REVIEW

File: addressable.gemspec [hunk 1]
```diff
@@ -6,30 +6,28 @@ Gem::Specification.new do |s|
...
+  s.files = ["CHANGELOG.md".freeze, "Gemfile".freeze, "LICENSE.txt".freeze, "README.md".freeze, "Rakefile".freeze, "data/unicode.data".freeze, "lib/addressable".freeze, "lib/addressable.rb".freeze, "lib/addressable/idna".freeze, "lib/addressable/idna.rb".freeze, "lib/addressable/idna/native.rb".freeze, "lib/addressable/idna/pure.rb".freeze, "lib/addressable/template.rb".freeze, "lib/addressable/uri.rb".freeze, "lib/addressable/version.rb".freeze, "spec/addressable".freeze, "spec/addressable/idna_spec.rb".freeze, "spec/addressable/net_http_compat_spec.rb".freeze, "spec/addressable/security_spec.rb".freeze, "spec/addressable/template_spec.rb".freeze, "spec/addressable/uri_spec.rb".freeze, "spec/spec_helper.rb".freeze, "tasks/clobber.rake".freeze, "tasks/gem.rake".freeze, "tasks/git.rake".freeze, "tasks/metrics.rake".freeze, "tasks/profile.rake".freeze, "tasks/rspec.rake".freeze, "tasks/yard.rake".freeze]
```
Comment (dentarg) [inline addressable.gemspec:16]: I think this change removed the following: `["addressable.gemspec", "benchmark", "coverage", "data", "documentation", "gemfiles", "lib", "spec", "specdoc", "tasks", "tmp"]`

Not sure how they got there in the first place?

Comment (dentarg) [reply addressable.gemspec:16]: Maybe use of newer RubyGems makes a difference

**Comment by petergoldstein** [comment]
@dentarg Once the gemspec updates are complete and any other changes are merged, would you be able to release an updated gem?  Thanks.

**Comment by dentarg** [comment]
Missed this comment but I replied at https://github.com/sporkmonger/addressable/pull/465#issuecomment-1194304433

**Pull request merged**